src/components/InlineAlert/InlineAlert.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { forwardRef, useEffect, useId, useRef } from 'react';
+import type { ReactElement, RefObject } from 'react';
+
+import { getIconColor, styles } from './InlineAlert.style';
+import type { InlineAlertProps } from './InlineAlert.types';
+import Icon from '../Icon';
+import { useDOMRef } from '../utils/useDOMRef';
+
+import useTheme from '~/hooks/useTheme';
+
+export const InlineAlert = forwardRef<HTMLDivElement, InlineAlertProps>(
+  (props: InlineAlertProps, ref: RefObject<HTMLDivElement>) => {
+    const {
+      status = 'neutral',
+      actions,
+      children,
+      onDismiss,
+      testId,
+      hasAutoFocus,
+      ...otherProps
+    } = props;
+    const theme = useTheme();
+    const dismissId = useId();
+    const domRef = useDOMRef(ref);
+
+    const actionElements =
+      actions && (actions as ReactElement).type === React.Fragment
+        ? (actions as ReactElement).props.children
+        : actions;
+    const actionsArray = React.Children.toArray(actionElements);
+
+    const autoFocusRef = useRef(hasAutoFocus);
+    useEffect(() => {
+      if (autoFocusRef.current && domRef.current) {
+        domRef.current.focus();
+      }
+      autoFocusRef.current = false;
+    }, [domRef]);
+
+    return (
+      <div
+        {...otherProps}
+        css={styles['inline-alert'](props)}
+        ref={domRef}
+        tabIndex={0}
+        role={status === 'warning' || status === 'error' ? 'alert' : 'status'}
+        aria-describedby={onDismiss ? dismissId : undefined}
+        data-slot="inline-alert"
+        data-testid={testId}
+      >
+        {status !== 'neutral' ? (
+          <Icon
+            role="img"
+            aria-hidden="true"
+            name={status}
+            css={styles.icon}
+            color={getIconColor(status, theme)}
+          />
+        ) : null}
+        <div css={styles.content}>{children}</div>
+        {actionsArray.length > 0 ? <div css={styles.actions}>{actionsArray}</div> : null}
+        {onDismiss ? (
+          <Icon
+            role="button"
+            aria-label="Dismiss notification"
+            id={onDismiss ? dismissId : undefined}
+            name="close"
+            css={styles.dismiss}
+            onClick={onDismiss}
+            size={20}
```

Comment:
is there a token for that? either comp or semantic?

Reply:
I followed the documentation from here `Interactive icon's size is a prop provided by the user; suggested sizes (as shown below) are 12 (xs), 16 (sm), 20 (m), 24 (l)`. Ideally we could add support for fixed sizes `sm`, `md` for the `InteractiveIcon` component at some point.

As for the `data-testid`, since we have specified aria attributes for the dismiss button, are you ok doing the following:
```js
const { container } = render(<InlineAlert onDismiss={onDismiss}>With dismiss</InlineAlert>);

const closeButton = within(container).getByRole('button', { name: 'Dismiss notification' });
```
It would be more preferred not to add a `dismissTestId` prop in this case, what do you think?

Reply:
you are right about the Icon size, I forgot the Interactive Icon's api tbh 😁 

about the test-id my only concern is that this might be confusing for the QA since they use data-testids regularly.

Reply:
Included `dismissTestId` on [4428743](https://github.com/Orfium/orfium-ictinus/pull/830/commits/4428743037fdc10742380e9c08e611c6e83bc77c)

Reply:
we have a `TestProps` type in `utils/types` that we add to all components for consistency, you can add this